### ISSUE TITLE
LocalDate generator values are missing maxDate

### ISSUE THREAD

**Issue by oliverblaha** [kotest/kotest#3874]
Hi,

From the docs of Arb.localDate():

> This generator creates randomly generated LocalDates, in the range [minDate, maxDate]

However, it only generates [minDate, maxDate), means maxDate will never be among the values.

The following snippet will only print 1999-12-31, but never 2000-01-01.
```
checkAll(Arb.localDate(minDate = LocalDate.of(1999, 12, 31), maxDate = LocalDate.of(2000, 1, 1))) {
    println(it)
}
```

Root cause is the same as for https://github.com/kotest/kotest/issues/3829. The fix for that issue unfortunately doesn't address the actual bug, but only works around it.

The actual fix should be this: `minDate.plusDays(it.random.nextLong(ChronoUnit.DAYS.between(minDate, maxDate) + 1))` in `dates.kt`
The added `+ 1` is required, because the argument to `nextLong` is the exclusive upper bound (aka the number of possible values).

So in the example above, the delta between the dates is `1`; `nextLong(1)` produces int values that are `>= 0` and `< 1`, so always `0`, which leads to the constant value of `minDate` being returned. Instead, you would want _two_ values (zero and one) to be returned by `nextLong`, therefore the upper exclusive bound must be `2`.

Adding the `+ 1` would not only fix this issue, but also the issue linked above; however, performance-wise it's nice to have a those extra checks, too.

Best,
Oliver

**Comment by Kantis** [comment]
Thanks, good catch. Would you like to raise a PR?

**Comment by oliverblaha** [comment]
Sure, here it is.
Note: maxDate _was_ actually covered by edge cases, but missing from regular samples.

**Comment by LeoColman** [comment]
Closed as per #3876
